src/loaders/jobs/taskfarmer/taskfarmer_task_mgr.py

```diff
@@ -0,0 +1,231 @@
+                return JobStatus.RUNNING
+            else:
+                return JobStatus.PENDING
+
+    def _append_task_info(self, task_info):
+        """
+        Append the task information to the task info file
+        """
+
+        task_info.update({'kbase_collection': self.kbase_collection, 'load_ver': self.load_ver, 'tool': self.tool})
+
+        if not all(key in task_info for key in REQUIRED_TASK_INFO_KEYS):
+            raise ValueError(f"task_info must contain all keys: {REQUIRED_TASK_INFO_KEYS}")
+
+        # Append the new record to the file
+        task_info_file = self._get_task_info_file()
+        with jsonlines.open(task_info_file, mode='a') as writer:
+            # lock the file to prevent multiple processes from writing to the same file
+            fcntl.flock(writer, fcntl.LOCK_EX)
+            writer.write(task_info)
+            fcntl.flock(writer, fcntl.LOCK_UN)
+
+    def __init__(self, kbase_collection, load_ver, tool, root_dir=loader_common_names.ROOT_DIR,
+                 destroy_old_job_dir=False):
+        """
+        Initialize the task manager.
+
+        :param kbase_collection: KBase collection identifier name (e.g. GTDB).
+        :param load_ver: collection load version (e.g. r207.kbase.1).
+        :param tool: tool name (e.g. gtdb_tk, checkm2, etc.)
+        :param root_dir: root directory of the collection project.
+                         Default is the ROOT_DIR defined in src/loaders/common/loader_common_names.py
+        :param destroy_old_job_dir: if True, remove contents of the old job directory.
+        """
+
+        self.kbase_collection = kbase_collection
+        self.load_ver = load_ver
+        self.tool = tool
+        self.root_dir = root_dir
+
+        # job directory is named as <kbase_collection>_<load_ver>_<tool>
+        self.job_dir = os.path.join(self.root_dir, tf_common.TASKFARMER_JOB_DIR,
+                                    f'{self.kbase_collection}_{self.load_ver}_{self.tool}')
+        self._task_exists = os.path.isdir(self.job_dir) and os.listdir(self.job_dir)
+        self._create_job_dir(destroy_old_job_dir=destroy_old_job_dir)
+        self._tasks_df = self._retrieve_all_tasks()
+
+    def get_latest_task(self):
+        """
+        Get the latest task information
+
+        Tasks are sorted by job_submit_time in descending order. The latest task is the first row.
+        """
+
+        return self._tasks_df.iloc[0].to_dict() if self._task_exists else {}
+
+    def retrieve_latest_task_status(self):
+        """
+        Retrieve the latest task status.
+
+        """
+        if not self._task_exists:
+            raise ValueError(f"Task does not exist for {self.kbase_collection}, {self.load_ver}, {self.tool}")
+
+        latest_task = self.get_latest_task()
+        job_status = self._get_job_status_from_nersc(latest_task["job_id"])
+
+        return job_status
+
+    def submit_job(self, source_data_dir):
+        """
+        Submit the job to slurm
```

Comment:
Should doc the requirements for the job to work, e.g. the taskfarmer script named X needs to be in the job dir

Reply:
👍

Reply:
:+1: